da/celestia/celestia.go

```diff
@@ -223,40 +229,106 @@ func (c *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultS
 				continue
 			}
 
-			//double check txResponse is not nil - not supposed to happen
-			if txResponse == nil {
-				err := errors.New("txResponse is nil")
-				c.logger.Error("Failed to submit DA batch", "error", err)
+			daMetaData := &da.DASubmitMetaData{
+				Height:      height,
+				Commitments: commitments,
+			}
+
+			availabilityResult := c.CheckBatchAvailability(daMetaData)
+			if availabilityResult.Code != da.StatusSuccess || !availabilityResult.DataAvailable {
+				c.logger.Error("Unable to confirm submitted blob availability. Retrying")
 				res, err := da.SubmitBatchHealthEventHelper(c.pubsubServer, c.ctx, false, err)
 				if err != nil {
 					return res
 				}
 				time.Sleep(c.submitRetryDelay)
```

Comment:
this shouldn't happen, because if the submission was successful, there should be valid proofs available for the blob. however, in case it happens, it makes sense to me to resubmit again, disregarding previous submission result, because from rpc calls it is not possible to know whether there was a problem in the submitted blob or in getting the proofs.